app/services/metrics/sidekiq_queue_sizes.rb

```diff
@@ -0,0 +1,40 @@
+module Metrics
+  class SidekiqQueueSizes
+    def self.call(prometheus_client)
+      new(prometheus_client).call
+    end
+
+    def initialize(prometheus_client)
+      @prometheus_client = prometheus_client
+    end
+
+    def call
+      queue_sizes.each do |queue, size|
+        send_metric(queue, size)
+      end
+    end
+
+    private
+
+    attr_reader :prometheus_client
+
+    def queue_sizes
+      @queue_sizes ||= queues.each_with_object({}) do |queue, hash|
+        hash[queue] = Sidekiq::Queue.new(queue.to_s).size
+      end
+    end
+
+    def send_metric(queue_name, size)
+      prometheus_client.send_json(
+        type: PrometheusCollectors::SidekiqQueueCollector::COLLECTOR_TYPE,
+        queue: queue_name,
+        size: size
+      )
+    end
```

Comment:
I thought it would be better to make all 3 calls to redis (queue_size) first and then 3 calls to the `metrics` container (send_metric) rather than alternate them but maybe it doesn't matter.

Reply:
It's a minor thing. There's a very small memory gain from missing out the middle hash and I think the code is a little simpler to read. But I have no problem with you keeping the original if you prefer it.